### PR TITLE
fix(docs): add missing changelog entries

### PR DESCRIPTION
**Pull request by mattculler**
- [ ] Have you followed the guidelines for contributing?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `make lint && make test`?

-----

Forgot to add to the changelog in #65.

### CODE REVIEW

File: docs/reference/changelog.rst [hunk 1]
```diff
@@ -2,6 +2,15 @@
 Changelog
 *********
 
+X.Y.Z (2024-MM-DD)
```
Comment (mr-cal) [inline docs/reference/changelog.rst:5]: Are you going to need a release of craft-platforms this pulse?

Comment (mattculler) [reply docs/reference/changelog.rst:5]: Yep, this snapcraft change will require a new craft-platforms release.

Comment (mattculler) [reply docs/reference/changelog.rst:5]: Yep that's fine.  But before we release craft-platforms let me finish the snapcraft change so I can be sure no more c-plat changes will be needed.

Comment (mattculler) [reply docs/reference/changelog.rst:5]: Don't need to make more changes to c-plat, we're all set on this one

Comment (lengau) [reply docs/reference/changelog.rst:5]: Change this to 0.4.0 with the date 2024-10-17 and I'll release in the morning!

**Review by lengau — APPROVED** [review]

File: docs/reference/changelog.rst (hunk 1 above)
Comment (mattculler) [reply docs/reference/changelog.rst:5]: Excellent, will do thanks!

**Review by tigarmo — APPROVED** [review]

**Pull request merged**